File: 1044-find-common-characters/1044-find-common-characters.cpp
class Solution {
public:
    vector<string> commonChars(vector<string>& words) {
        unordered_map<char, int> base, counter; 
        vector<string> Final; 
        for(char c: words[0]){
                base[c]++; 
            }

        for(int i = 1; i< words.size(); i++){
            for(char c: words[i]){
                if(base.count(c) && base[c] > counter[c]){
                    counter[c]++; 
                }
            }
            base = counter; 
            counter.clear(); 
        }

        for(auto& c : base){
            for(int j = 0; j < c.second; j++){
                Final.push_back(string(1, c.first)); 
            }
        }

        return Final; 
        
    }
};

File: 1044-find-common-characters/1044-find-common-characters.py
class Solution:
    def commonChars(self, words: List[str]) -> List[str]:
        cnt = Counter(words[0])

        for i in range(1, len(words)): 
            curr_cnt = Counter(words[i])
            for c in cnt: 
                cnt[c] = min(cnt[c], curr_cnt[c])
        
        res = []
        for key in cnt:
            for i in range(cnt[key]): 
                res.append(key)
        return res 